static/js/components/OrderSummary.js

```diff
@@ -68,15 +68,18 @@ class OrderSummary extends React.Component {
   }
 
   render() {
-    let { course, courseRun, checkout, checkoutStatus, couponCode} = this.props;
+    let { course, courseRun, checkout, checkoutStatus, couponCode, discount} = this.props;
     let discountInfo;
-    if (couponCode) {
+    const message = couponCode ? `Discount from coupon ${couponCode}` : 'Discount from coupon';
```

Comment:
This line should be moved inside the `if (discount) {` block since it's only used there